File: bratek20-spring/src/main/java/pl/bratek20/spring/web/WebAppConfig.java
package pl.bratek20.spring.web;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;
import pl.bratek20.spring.config.BaseConfig;

@Configuration
@Import({
    BaseConfig.class,
})
class WebAppConfig {
    @Bean
    public HealthController healthController() {
        return new HealthController();
    }
}

File: bratek20-spring/src/main/kotlin/com/github/bratek20/spring/webapp/WebAppConfig.kt
package com.github.bratek20.spring.webapp

import com.github.bratek20.infrastructure.httpserver.api.PassingExceptionHandler
import com.github.bratek20.logs.api.Logger
import com.github.bratek20.spring.config.BaseConfig
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import

@Configuration
@Import(
    BaseConfig::class,
)
internal open class WebAppConfig {
    @Bean
    open fun healthController(): HealthController {
        return HealthController()
    }

    @Bean
    open fun passingExceptionHandler(logger: Logger): PassingExceptionHandler {
        return PassingExceptionHandler(logger)
    }
}
